lib/iris/_data_manager.py

```diff
@@ -382,6 +411,26 @@ def dtype(self):
         return result
 
     @property
+    def fill_value(self):
+        return self._fill_value
+
+    @fill_value.setter
+    def fill_value(self, fill_value):
+        if fill_value is not None:
+            # Convert the given value to the dtype of the data manager.
+            fill_value = np.asarray([fill_value])[0]
+            target_dtype = self.dtype
+            if fill_value.dtype.kind == 'f' and target_dtype.kind in 'biu':
+                # Perform rounding when converting floats to ints.
+                fill_value = np.rint(fill_value)
```

Comment:
@pp-mo 
> if you rounded down then converting real-->lazy-->real would not preserve the data values of masked integer data.

I'm puzzled by this. Isn't it the case that casting an int to a float results in either the same value, or some other integer? If that's the case then whether you round up or down when converting back should make no difference.